repoaudit/repoaudit/utils.py

```diff
@@ -213,6 +215,60 @@ def check_signature(repo: str, dist: str, file_url: str,
     return True
 
 
+def verify_checksum(
+    repo: str,
+    dist: str,
+    file_loc_in_repo: str,
+    file_type: str,
+    expected_checksums: List[Tuple[str, str]],
+    errors: RepoErrors,
+    error_if_missing: bool = True,
+    verify: Optional[str] = None
+) -> bool:
+    """
+    Verify the checksum of a file in a repository. file_loc_in_repo is the
+    location of the file relative to the repository url. file_type indicates
+    whether a "metadata" or "package" file is being checked and is used for
+    error messages. expected_checksums is an array of tuples with the format
+    [(checksum_algorithm, expected_checksum), ...]. If error_if_missing is
+    set to False, then no error will be reported if the file is missing.
+    """
+    try:
+        response = get_url(urljoin(repo, file_loc_in_repo), verify=verify, stream=True)
+    except HTTPError as e:
+        if error_if_missing:
+            errors.add(
+                repo, dist,
+                f"Could not access {file_type} file at {e.response.url}: {e}"
+            )
+        return not error_if_missing
+
+    checksum_algorithms = [elem[0] for elem in expected_checksums]
+    multihash = MultiHash(checksum_algorithms)
+
+    for chunk in response.iter_content(chunk_size=CHUNK_SIZE):
+        multihash.update(chunk)
+
+    success = True
+    error_str = f"{file_type.capitalize()} checksum mismatch for '{file_loc_in_repo}': ["
+
+    for alg, expected in expected_checksums:
+        if multihash.hexdigest(alg) != expected:
+            comma = '' if success else ', '
+            error_str += (
+                f"{comma}{alg.upper()} expected '{expected}' "
+                f"but received '{multihash.hexdigest(alg)}'"
+            )
```

Comment:
The pythonic way to do this is to build a list of individual error message fragments and then use join, i.e.
```
parts = []
for alg, expected...
    if multihash.hexdigest...
        parts.append(f"{alt.upper()} expected ... but received ...")
if parts:
    error_str = f"... checksum mismatch for '{file...}': [{", ".join(parts)}]"
    errors.add(repo, dist, error_str)
    return False

return True
```

Reply:
Yeah that's cleaner. I changed to code to this